### DISCUSSION TITLE
Element as Text best method

### DISCUSSION THREAD

**Discussion by Kashumy** [community/community#60292] [Programming Help]
### Select Topic Area

Question

### Body

# Code Html
i found a method to put an html element as text.
but i am looking for a better method

````
<pre  style="white-space: pre-wrap; margin: 10px;"><script>var element = document.currentScript.parentElement;
  element.innerText = `
 <h1></h1>
  `;
</script></pre>

````
does anyone know of an optimized method that doesn't need libraries either

**Comment by nirav-gajera** [comment]
If you're looking for a more optimized and lightweight method to set the content of an HTML element without using any libraries, you can directly access the element using its ID or query selector and update its `innerHTML` property

HTML: `<div id="myElement"></div>
`

JS: `var element = document.getElementById('myElement');
element.innerHTML = '<h1>My Heading</h1>';
`

or

HTML: `<div id="myElement"></div>
`

JS: `var element = document.getElementById('myElement');
var heading = document.createElement('h1');
heading.textContent = 'My Heading';
element.appendChild(heading);
`

**Reply by Kashumy** [reply]
@nirav-gajera    it's also a great idea but it's more so if you want to boast that you're making a website in JS.
I am looking for a method that is not resource-intensive, it is always some kilobytes
I'm looking for a method that will allow me to write less script

**Reply by nirav-gajera** [reply]
```
<div id="myElement"></div>
var element = document.getElementById('myElement');
element.innerHTML = `
  <h1>My Heading</h1>
`;

```
